stdlibs/stdlibs.go

```diff
@@ -21,6 +24,59 @@ func InjectNativeMappings(store gno.Store) {
 
 func InjectPackage(store gno.Store, pn *gno.PackageNode) {
 	switch pn.PkgPath {
+	case "internal/crypto/sha3":
+		pn.DefineNative("Sum256",
+			gno.Flds( // params
+				"data", "[]byte",
+			),
+			gno.Flds( // results
+				"bz", "[]byte",
+			),
+			func(m *gno.Machine) {
+				arg0 := m.LastBlock().GetParams1().TV
+				bz := []byte(nil)
+				if arg0.V != nil {
+					slice := arg0.V.(*gno.SliceValue)
+					array := slice.GetBase(m.Store)
+					bz = array.GetReadonlyBytes()[:slice.Length]
+				}
+				// remove padding
+				bzNP := bytes.Trim(bz, "\x00")
```

Comment:
Why are bytes padded with zero bytes in the first place here? It seems wrong to do this... (we should hash alll bytes in the input)

Reply:
that was an old legacy when I didn't slice the result of GetReadonlyBytes().
just forgot to remove it :D